### PR TITLE
Drop cli unknown options and others improvements.

### PR DESCRIPTION
**Pull request by mshima**
- ~Creates a local copy of commander.js 7 snapshot from https://github.com/tj/commander.js/blob/b4fbe678efcf3c51fec2b7710fe529eaa0b7e0ec/index.js~
- Updated commander to 7.0.0-0 pre-release.
Required due to updated apis for new features.
- Load generator arguments definitions from the generator itself, instead of a additional definition at commands.js
- Lazily load options for startup speed improvement.
- Fixes options definitions.
- Cleanups.

Related to https://github.com/jhipster/generator-jhipster/issues/11896

~Waiting for https://github.com/jhipster/generator-jhipster/pull/12703 to be reviewed.~

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@pascalgrimaud it's done, should I wait for a review?

**Comment by pascalgrimaud** [comment]
Too complex for me.
Hope @MathieuAA will have the time to have a quick look
Otherwise, I'll trust the CI :)

File: cli/utils.js [hunk 1]
```diff
@@ -161,34 +143,17 @@ const getOptionAsArgs = (options = {}) => {
 /**
  *  Get options for the command
  */
-const getCommand = (cmd, args, opts) => {
-    let options = [];
-    if (opts && opts.argument && opts.argument.length > 0) {
+const getCommand = (cmd, args = []) => {
+    let cmdArgs;
+    if (args && args.length > 0) {
```
Comment (MathieuAA) [inline cli/utils.js:148]: args is always set, no need for this check

**Review by MathieuAA — APPROVED** [review]

File: cli/utils.js (hunk 1 above)
Comment (mshima) [reply cli/utils.js:148]:
```suggestion
    if (args.length > 0) {
```

**Comment by mshima** [comment]
Thanks @MathieuAA

**Pull request merged**